packages/core/src/Platform/index.ts

```diff
@@ -1,30 +1,37 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
+import { Platform as PlatformType, CustomUserAgent } from '../types/types';
 import { version } from './version';
+import { detectFramework } from './detectFramework';
 
-const BASE_USER_AGENT = `aws-amplify/${version}`;
+const BASE_USER_AGENT = 'aws-amplify';
 
-export const Platform = {
-	userAgent: `${BASE_USER_AGENT} js`,
+export const Platform: PlatformType = {
+	userAgent: `${BASE_USER_AGENT}/${version}`,
 	product: '',
 	navigator: null,
 	isReactNative: false,
 };
 if (typeof navigator !== 'undefined' && navigator.product) {
 	Platform.product = navigator.product || '';
 	Platform.navigator = navigator || null;
-	switch (navigator.product) {
-		case 'ReactNative':
-			Platform.userAgent = `${BASE_USER_AGENT} react-native`;
-			Platform.isReactNative = true;
-			break;
-		default:
-			Platform.userAgent = `${BASE_USER_AGENT} js`;
-			Platform.isReactNative = false;
-			break;
+	if (navigator.product === 'ReactNative') {
+		Platform.isReactNative = true;
 	}
 }
 
-export const getAmplifyUserAgent = (content?: string) => {
-	return `${Platform.userAgent}${content ? content : ''}`;
+export const getAmplifyUserAgent = (customUserAgent?: CustomUserAgent) => {
+	return `${Platform.userAgent} ${buildUserAgentDetails(customUserAgent)}`;
+};
+
+const buildUserAgentDetails = (customUserAgent?: CustomUserAgent) => {
+	let userAgentDetails = customUserAgent;
+
+	if (userAgentDetails) {
+		userAgentDetails.framework =
+			userAgentDetails.framework ?? detectFramework();
+	} else {
+		userAgentDetails = { framework: detectFramework() };
+	}
```

Comment:
Ditto here about mutating the param. We should be able to use the same pattern as well of spreading the param - since `detectFramework` isn't exactly a compute heavy function.
```suggestion
	const userAgentDetails = { framework: detectFramework(), ...userAgentDetails }
```